packages/backend-function/src/factory.test.ts

```diff
@@ -304,6 +304,75 @@ void describe('AmplifyFunctionFactory', () => {
     });
   });
 
+  void describe('environment property', () => {
+    void it('sets valid environment', () => {
+      const functionFactory = defineFunction({
+        entry: './test-assets/default-lambda/handler.ts',
+        name: 'myCoolLambda',
+        environment: {
+          TEST_ENV: 'testValue',
+        },
+      });
+      const lambda = functionFactory.getInstance(getInstanceProps);
+      const stack = lambda.stack;
+      const template = Template.fromStack(stack);
+      template.resourceCountIs('AWS::Lambda::Function', 1);
+      template.hasResourceProperties('AWS::Lambda::Function', {
+        Environment: {
+          Variables: {
+            TEST_ENV: 'testValue',
+          },
+        },
+      });
+    });
+
+    void it('sets default environment', () => {
+      const functionFactory = defineFunction({
+        entry: './test-assets/default-lambda/handler.ts',
+        name: 'myCoolLambda',
+      });
+      const lambda = functionFactory.getInstance(getInstanceProps);
+      const stack = lambda.stack;
+      const template = Template.fromStack(stack);
+      template.resourceCountIs('AWS::Lambda::Function', 1);
+      template.hasResourceProperties('AWS::Lambda::Function', {
+        Environment: {},
+      });
+    });
+
+    void it('throws when adding environment variables with invalid key', () => {
+      assert.throws(
+        () =>
+          defineFunction({
+            entry: './test-assets/default-lambda/handler.ts',
+            name: 'myCoolLambda',
+            environment: {
+              ['this.is.wrong']: 'testValue',
```

Comment:
a suggestion:
it might be worth running this in the loop with couple of wrong values.

for example `['a', 'this.is.wrong', ...]`

reason is that. - we removed that length check, but we should still validate that regex catches 1 character violation.

Reply:
I have that captured in another [unit test](https://github.com/aws-amplify/amplify-backend/pull/2290/files#diff-52b21477dc0ed703b86e698d1ebae4e5705509e1edf1ee2dbfaa9f8befd7bafeR359). But this got me thinking if we should validate all keys and list them in the error?

Reply:
> But this got me thinking if we should validate all keys and list them in the error?

that's better DX.